src/Core/Resolvers/QueryExecutor.cs

```diff
@@ -636,6 +645,39 @@ DbResultSet dbResultSet
             return resultProperties;
         }
 
+        /// <summary>
+        /// Reads data into jsonString.
+        /// </summary>
+        /// <param name="dbDataReader">DbDataReader.</param>
+        /// <param name="availableSize">Available buffer.</param>
+        /// <param name="jsonString">jsonString to read into.</param>
+        /// <returns>size of data read in bytes.</returns>
+        public int StreamData(DbDataReader dbDataReader, int availableSize, StringBuilder jsonString)
+        {
+            int fieldSize = (int)dbDataReader.GetChars(0, 0, null, 0, 0);
+
+            // if the size of the field is less than available size, then we can read the entire field.
+            // else we throw exception.
+            ValidateSize(availableSize, fieldSize);
+
+            char[] buffer = new char[fieldSize];
+
+            // read entire field into buffer and reduce available size.
+            dbDataReader.GetChars(0, 0, buffer, 0, fieldSize);
+
+            jsonString.Append(buffer[0..(int)(fieldSize)]);
+            return fieldSize;
+        }
+
+        /// <summary>
+        /// This function reads the data from the DbDataReader and returns the JSON string.
+        /// 1. MaxResponseSizeLogicEnabled is used like a feature flag.
+        /// 2. If MaxResponseSize is not specified by the customer or is null, getString is used and entire data is read into memory.
+        /// 3. If MaxResponseSize is specified by the customer, getChars is used. GetChars tries to read the data in chunks and if the data is more than the specified limit, it throws an exception.
```

Comment:
nit: break up these lines. try to keep ~120 -130 char width per line.

Reply:
fixed.